src/pymor/bindings/dunegdt.py

```diff
@@ -0,0 +1,230 @@
+            https://zivgitlab.uni-muenster.de/ag-ohlberger/dune-community/dune-xt/-/tree/master/dune/xt/la/solver
+            for available options, depending on the type of `matrix`. E.g., for
+            dune.xt.la.IstlSparseMatrix, (as can be queried from dune.xt.la.IstlSparseMatrixSolver
+            via `types()` and `options(type)`):
+              - 'bicgstab.ssor'
+              - 'bicgstab.amg.ssor'
+              - 'bicgstab.amg.ilu0'
+              - 'bicgstab.ilut'
+              - 'bicgstab'
+              - 'cg'
+        name
+            Optional name of the resulting |Operator|.
+        """
+
+        linear = True
+
+        def __init__(self, matrix, source_id='STATE', range_id='STATE', solver_options=None, name=None):
+            self.source = DuneXTVectorSpace(matrix.cols, matrix.vector_type(), source_id)
+            self.range = DuneXTVectorSpace(matrix.rows, matrix.vector_type(), range_id)
+            self.__auto_init(locals())
+
+        def _real_apply_one_vector(self, u, mu=None, prepare_data=None):
+            r = self.range.real_zero_vector()
+            self.matrix.mv(u.impl, r.impl)
+            return r
+
+        def _apply_adjoint_one_vector(self, v, mu=None, prepare_data=None):
+            r = self.source.real_zero_vector()
+            self.matrix.mtv(v.impl, r.impl)
+            return r
+
+        def _real_apply_inverse_one_vector(self, v, mu=None, initial_guess=None,
+                                           least_squares=False, prepare_data=None):
+            if least_squares:
+                raise NotImplementedError
+            r = (self.source.real_zero_vector() if initial_guess is None else
+                 initial_guess.copy(deep=True))
+            options = self.solver_options.get('inverse') if self.solver_options else None
+
+            from dune.xt.la import make_solver
+            solver = make_solver(self.matrix)
+            if options:
+                solver.apply(v.impl, r.impl, options)
+            else:
+                solver.apply(v.impl, r.impl)
+            return r
+
+        def _assemble_lincomb(self, operators, coefficients, identity_shift=0., solver_options=None, name=None):
+            if not all(isinstance(op, DuneXTMatrixOperator) for op in operators):
+                return None
+            if identity_shift != 0:
+                return None
+            if np.iscomplexobj(coefficients):
+                return None
+
+            if coefficients[0] == 1:
+                matrix = operators[0].matrix.copy()
+            else:
+                matrix = operators[0].matrix * coefficients[0]
+            for op, c in zip(operators[1:], coefficients[1:]):
+                matrix.axpy(c, op.matrix)  # TODO: Not guaranteed to work for all backends! For different
+                # sparsity patterns one would have to extract the patterns from the pruned
+                # matrices, merge them and create a new matrix.
```

Comment:
I hope there will be an error in case of incompatible sparsity patterns?

Reply:
Actually, a quick test revealed that it works by now, might be that we added some conversion to dune-xt and I forgot, or that dune-istl improved. I would leave the comment here anyway in case some dune backend raises an error. It should then pop up in the stack trace and we know where to start looking.